services/azure-devops/azure-devops-coverage.service.js

```diff
@@ -0,0 +1,168 @@
+const { buildOptions } = require('./azure-devops-helpers')
+
+const documentation = `
+<p>
+  To obtain your own badge, you need to get 3 pieces of information:
+  <code>ORGANIZATION</code>, <code>PROJECT</code> and <code>DEFINITION_ID</code>.
+</p>
+<p>
+  First, you need to select your build definition and look at the url:
+</p>
+<img
+  src="https://user-images.githubusercontent.com/3749820/47259976-e2d9ec80-d4b2-11e8-92cc-7c81089a7a2c.png"
+  alt="ORGANIZATION is after the dev.azure.com part, PROJECT is right after that, DEFINITION_ID is at the end after the id= part." />
+<p>
+  Your badge will then have the form:
+  <code>https://img.shields.io/azure-devops/coverage/ORGANIZATION/PROJECT/DEFINITION_ID.svg</code>.
+</p>
+<p>
+  Optionally, you can specify a named branch:
+  <code>https://img.shields.io/azure-devops/coverage/ORGANIZATION/PROJECT/DEFINITION_ID/NAMED_BRANCH.svg</code>.
+</p>
+`
+const {
+  coveragePercentage: coveragePercentageColor,
+} = require('../../lib/color-formatters')
+
+const latestBuildSchema = Joi.object({
+  count: Joi.number().required(),
+  value: Joi.array()
+    .items(
+      Joi.object({
+        id: Joi.number().required(),
+      })
+    )
+    .required(),
+}).required()
+
+const buildCodeCoverageSchema = Joi.object({
+  coverageData: Joi.array()
+    .items(
+      Joi.object({
+        coverageStats: Joi.array()
+          .items(
+            Joi.object({
+              label: Joi.string().required(),
+              total: Joi.number().required(),
+              covered: Joi.number().required(),
+            })
+          )
+          .min(1)
+          .required(),
+      })
+    )
+    .required(),
+}).required()
+
+module.exports = class AzureDevOpsCoverage extends BaseJsonService {
+  static render({ coverage }) {
+    return {
+      message: `${coverage.toFixed(0)}%`,
+      color: coveragePercentageColor(coverage),
+    }
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'coverage' }
+  }
+
+  static get category() {
+    return 'build'
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Azure DevOps coverage',
+        urlPattern:
+          'azure-devops/coverage/:organization/:project/:definitionId',
+        staticExample: this.render({ coverage: 100 }),
+        exampleUrl: 'azure-devops/coverage/swellaby/opensource/25',
+        keywords: ['vso', 'vsts', 'azure-devops'],
+        documentation,
+      },
+      {
+        title: 'Azure DevOps coverage (branch)',
+        urlPattern:
+          'azure-devops/coverage/:organization/:project/:definitionId/:branch',
+        staticExample: this.render({ coverage: 100 }),
+        exampleUrl: 'azure-devops/coverage/swellaby/opensource/25/master',
+        keywords: ['vso', 'vsts', 'azure-devops'],
+        documentation,
+      },
+    ]
+  }
+
+  static get route() {
+    return {
+      base: '',
+      format: '(?:azure-devops|vso)/coverage/([^/]+)/([^/]+)/([^/]+)(?:/(.+))?',
```

Comment:
Since this is a new badge, let's not support the `vso/` alias. That way you can move `azure-devops` to `base` and remove references to it elsewhere.

Reply:
Makes sense, will do